hibernate-core/src/main/java/org/hibernate/NaturalIdMultiLoadAccess.java

```diff
@@ -0,0 +1,130 @@
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	NaturalIdMultiLoadAccess<T> with(CacheMode cacheMode);
+
+	/**
+	 * Define a load graph to be used when retrieving the entity
+	 */
+	default NaturalIdMultiLoadAccess<T> with(RootGraph<T> graph) {
+		return with( graph, GraphSemantic.LOAD );
+	}
+
+	/**
+	 * Define a load or fetch graph to be used when retrieving the entity
+	 */
+	NaturalIdMultiLoadAccess<T> with(RootGraph<T> graph, GraphSemantic semantic);
+
+	/**
+	 * Specify a batch size for loading the entities (how many at a time).  The default is
+	 * to use a batch sizing strategy defined by the Dialect in use.  Any greater-than-one
+	 * value here will override that default behavior.  If giving an explicit value here,
+	 * care should be taken to not exceed the capabilities of the underlying database.
+	 * <p/>
+	 * Note that overall a batch-size is considered a hint.
+	 *
+	 * @param batchSize The batch size
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	NaturalIdMultiLoadAccess<T> withBatchSize(int batchSize);
+
+	/**
+	 * Should the multi-load operation be allowed to return entities that are locally
+	 * deleted?  A locally deleted entity is one which has been passed to this
+	 * Session's {@link Session#delete} / {@link Session#remove} method, but not
+	 * yet flushed.  The default behavior is to handle them as null in the return
+	 * (see {@link #enableOrderedReturn}).
+	 *
+	 * @param enabled {@code true} enables returning the deleted entities;
+	 * {@code false} (the default) disables it.
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	NaturalIdMultiLoadAccess<T> enableReturnOfDeletedEntities(boolean enabled);
+
+	/**
+	 * Should the return List be ordered and positional in relation to the
+	 * incoming ids?  If enabled (the default), the return List is ordered and
+	 * positional relative to the incoming ids.  In other words, a request to
+	 * {@code multiLoad([2,1,3])} will return {@code [Entity#2, Entity#1, Entity#3]}.
+	 * <p/>
+	 * An important distinction is made here in regards to the handling of
+	 * unknown entities depending on this "ordered return" setting.  If enabled
+	 * a null is inserted into the List at the proper position(s).  If disabled,
+	 * the nulls are not put into the return List.  In other words, consumers of
+	 * the returned ordered List would need to be able to handle null elements.
+	 *
+	 * @param enabled {@code true} (the default) enables ordering;
+	 * {@code false} disables it.
+	 *
+	 * @return {@code this}, for method chaining
+	 */
+	NaturalIdMultiLoadAccess<T> enableOrderedReturn(boolean enabled);
+
+	/**
+	 * Perform a load of multiple entities by natural-id.
+	 *
+	 * See {@link #enableOrderedReturn} and {@link #enableReturnOfDeletedEntities}
+	 * for options which effect the size and "shape" of the return list.
+	 *
+	 * @param ids The natural-id values to load
+	 *
+	 * @return The managed entities.
+	 */
+	List<T> multiLoad(Object... ids);
```

Comment:
So the above method implies that primitive array (like `int[]` or `long[]`) cannot be passed in directly. Luckily, we have a util method (`toObjectArray(Object array)`) created by me in ArrayHelper for the transformation.
It might be worthwhile considering providing overloaded methods accepting primitive type array, if for completeness sake.